src/systems/userinput/devices/gear-vr-controller.js

```diff
@@ -0,0 +1,50 @@
+import { paths } from "../paths";
+import { Pose } from "../pose";
+
+export class GearVRControllerDevice {
+  constructor(gamepad) {
+    this.gamepad = gamepad;
+    this.buttonMap = [{ name: "touchpad", buttonId: 0 }, { name: "trigger", buttonId: 1 }];
+    this.axisMap = [{ name: "touchpadX", axisId: 0 }, { name: "touchpadY", axisId: 1 }];
+
+    this.rayObjectRotation = new THREE.Quaternion();
+    this.selector = `#player-${gamepad.hand}-controller`;
+    this.pose = new Pose();
+  }
+
+  write(frame) {
+    if (this.gamepad.connected) {
+      this.gamepad.buttons.forEach((button, i) => {
+        const buttonPath = paths.device.gamepad(this.gamepad.index).button(i);
+        frame[buttonPath.pressed] = !!button.pressed;
+        frame[buttonPath.touched] = !!button.touched;
+        frame[buttonPath.value] = button.value;
+      });
+      this.gamepad.axes.forEach((axis, i) => {
+        frame[paths.device.gamepad(this.gamepad.index).axis(i)] = axis;
+      });
+
+      this.buttonMap.forEach(button => {
+        const outpath = paths.device.gearVRController.button(button.name);
+        frame[outpath.pressed] = !!frame[paths.device.gamepad(this.gamepad.index).button(button.buttonId).pressed];
+        frame[outpath.touched] = !!frame[paths.device.gamepad(this.gamepad.index).button(button.buttonId).touched];
+        frame[outpath.value] = frame[paths.device.gamepad(this.gamepad.index).button(button.buttonId).value];
+      });
+      this.axisMap.forEach(axis => {
+        frame[paths.device.gearVRController.axis(axis.name)] =
+          frame[paths.device.gamepad(this.gamepad.index).axis(axis.axisId)];
+      });
+
+      // TODO ideally we should just be getting pose from the gamepad
```

Comment:
We can if we just add in an elbow model like in
 https://github.com/aframevr/aframe/blob/7f9c5bffeaa67a6b78b35f1401e336019914ac3b/src/components/tracked-controls.js#L130
or
https://github.com/MozillaReality/FirefoxReality/blob/b019f45785fbb4bee534811ffa38aa27509915f3/app/src/main/cpp/ElbowModel.cpp